File: programmers/lv2/영어_끝말잇기.py
def solution(n, words):
    word = set([words[0]])
    num = 0
    order = 0
    for i in range(1, len(words)):
        if words[i] in word or words[i][0] != words[i-1][-1]:
            num = i % n + 1
            order = i // n + 1
            break
        else:
            word.add(words[i])
    return [num, order]

File: programmers/lv2/영어_끝말잇기.js
function solution(n, words) {
  const existing = [];
  for (let i = 0; i < words.length; i++) {
    if (
      existing.includes(words[i]) ||
      words[i].length === 1 ||
      (i !== 0 && words[i - 1][words[i - 1].length - 1] !== words[i][0])
    ) {
      return [(i % n) + 1, parseInt(i / n) + 1];
    }
    existing.push(words[i]);
  }
  return [0, 0];
}
